osbuild/loop.py

```diff
@@ -126,6 +126,8 @@ def __init__(self, minor, dir_fd=None):
             if not dir_fd:
                 dir_fd = os.open("/dev", os.O_DIRECTORY)
                 stack.callback(lambda: os.close(dir_fd))
+            if not os.path.exists(os.path.join("/dev", self.devname)):
```

Comment:
Hmm, the dfd-relative-ness of the surrounding code makes it seem like it's possible for `dir_fd` to not refer to `/dev`. So we should probably use a dfd-relative API here too. E.g.

```python
try:
    os.stat(self.devname, dfd=dir_fd)
except FileNotFound:
    self.mknod(dir_fd)
```

Reply:
done